### DISCUSSION TITLE
Building for 4.22

### DISCUSSION THREAD

**Discussion by LMoneyshot** [microdee/OpenUnrealSpaceMouse#66] [Q&A]
Hi.

I asked about being able to build this for ue4 4.22, and got recommended to try to build it locally since it wasn't supported officially.

This last command:
".\build.cmd --target MakeRelease --unreal-version 4.22.3"
Gives me this error: error msb3644
"The reference assemblies for framework ".NETFramework ,Version=V5.0" were not found.  To resolve this, install the SDK or Targeting Pack for this framework version or retarget your application to a version of the framework for which you have the SDK or Targeting Pack installed."

Any idea on how I should proceed?

**Comment by microdee** [comment]
https://dotnet.microsoft.com/download/dotnet/5.0

You might also have to install Visual Studio 2019 if you haven't already (not Visual Studio Code!) with C++ and .NET development loadouts.

**Reply by LMoneyshot** [reply]
I'm getting help from a friend and right now we're at the point of MakeRelease command but we're facing a new error:

C:\Program Files\dotnet\sdk\5.0.403\Roslyn\Microsoft.CSharp.Core.targets(71,5): error : Unhandled exception.  [I:\UE4-SpaceMouse-TestProject\UE4-SpaceMouse-TestProject\Nuke.Targets\_build.csproj]
C:\Program Files\dotnet\sdk\5.0.403\Roslyn\Microsoft.CSharp.Core.targets(71,5): error :    Cannot print exception string because Exception.ToString() failed. [I:\UE4-SpaceMouse-TestProject\UE4-SpaceMouse-TestProject\Nuke.Targets\_build.csproj]

2019+net5.0 installed

**Reply by microdee** [reply]
I'd need more context but just a hunch: maybe submodules are not checked out correctly (submodules are self-contained git repositories nested inside a git repository):
1. a: if you downloaded the repository as zip, delete that repository and clone it via git:
  ```
  git clone https://github.com/microdee/UE4-SpaceMouse-TestProject.git --recurse-submodules
  ```
1. b: if you have cloned the repo do
  ```
  git submodule update --init --recursive
  ```
2. Make sure that the `Nuke.Unreal` and `Plugins/SpaceMouse` folder is not empty or it has more than a `.git` file in them (these are git submodules).
3. If it was cloned properly before and submodules are there then go to both the `Nuke.Unreal` and the `Plugins/SpaceMouse` folders and do
  ```
  git switch master
  git pull
  ```
4. If those doesn't work, I have no idea what can cause that problem unless you copy/paste more of the output log. If that won't help then you'll need to include and build this plugin in your project.

**Reply by LMoneyshot** [reply]
It seems it was a matter of UE4 engine install path. It got built, but with some lines of errors.
I'm able to get smooth movement if I go slow. If I go any faster it skips ahead, like a rubberband and movement on the xyz is very slow and unaffected by speed value changes.

**Reply by microdee** [reply]
> I'm able to get smooth movement if I go slow. If I go any faster it skips ahead, like a rubberband and movement on the xyz is very slow and unaffected by speed value changes.

for those problems please check the **[Known Issues](https://github.com/microdee/UE4-SpaceMouse/blob/master/README.md#known-issues)** section of the SpaceMouse readme.md https://github.com/microdee/UE4-SpaceMouse/blob/master/README.md

**Reply by LMoneyshot** [reply]
That was it! Figured it was all of the errors..!

So then it's confirmed it works for 4.22.3 as well. Thank you very much!
